src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs

```diff
@@ -170,7 +170,7 @@ public void End(bool captureResponseTime = true)
             };
 
             // The completion of transactions can be run on thread or off thread. We made this configurable.  
-            if (Agent.Configuration.CompleteTransactionsOnThread)
+            if (Agent.Configuration.CompleteTransactionsOnThread || Agent.Configuration.ServerlessModeEnabled)
```

Comment:
Would it be better to move the ServerlessModeEnabled check into the `CompleteTransactionsOnThread` property in DefaultConfiguration?

Reply:
If we move the logic int the CompleteTransactionsOnThread property, we can more easily unit test that logic.

Reply:
I considered that but I was concerned it would hide the fact that serverless mode was behaving differently. I don't feel too strongly though if you think the override is cleaner.

Reply:
Either approach works for me (I don't think that we have consistency in the code base either way), but I think the logic should be covered in a unit test.